lib/gnat.ex

```diff
@@ -24,6 +24,82 @@ defmodule Gnat do
   }
   @type sent_message :: {:msg, message()}
 
+  @typedoc """
+  * `connection_timeout` - limits how long it can take to establish a connection to a server
+  * `host` - The location of the NATS server
+  * `port` - The port the NATS server is listening on
+  * `ssl_opts` - Options for connecting over SSL
+  * `tcp_opts` - Options for connecting over TCP
+  * `tls` - If the server should use a TLS connection
+  * `inbox_prefix` - Prefix to use for the message inbox of this connection
+  """
+  @type connection_settings :: %{
```

Comment:
😍 yes! I love that you documented this as a typespec with typeddoc. Do you want to update the typespec for the `start_link` function to specify that it accepts `connection_settings()` instead of just a generic `map()`?

Reply:
Good idea. Updated that here [update start_link](https://github.com/nats-io/nats.ex/pull/124/commits/72cd0ccbccd5d01defe2cb179eb42427ab20cd1a)